From 74d2e1e85f9131e7a7bb0e07aea46759550b4f7b Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Tue, 3 Sep 2024 12:52:12 -0400
Subject: [PATCH] Update find vigra

---
 cmake/modules/FindVIGRA.cmake | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/cmake/modules/FindVIGRA.cmake b/cmake/modules/FindVIGRA.cmake
index 7176ea6e..a5247ca8 100644
--- a/cmake/modules/FindVIGRA.cmake
+++ b/cmake/modules/FindVIGRA.cmake
@@ -13,18 +13,22 @@ FIND_LIBRARY(VIGRA_IMPEX_LIBRARY vigraimpex PATHS $ENV{VIGRA_ROOT}/src/impex $EN
 GET_FILENAME_COMPONENT(VIGRA_IMPEX_LIBRARY_PATH ${VIGRA_IMPEX_LIBRARY} PATH)
 SET( VIGRA_IMPEX_LIBRARY_DIR ${VIGRA_IMPEX_LIBRARY_PATH} CACHE PATH "Path to Vigra impex library.")
 
-EXECUTE_PROCESS ( COMMAND python -c "from distutils.sysconfig import get_python_lib; print get_python_lib()" OUTPUT_VARIABLE PYTHON_SITE_PACKAGES OUTPUT_STRIP_TRAILING_WHITESPACE)
-FIND_FILE(VIGRA_NUMPY_CORE_LIBRARY 
+find_package(Python 3 COMPONENTS NumPy)
+FIND_FILE(VIGRA_NUMPY_CORE_LIBRARY
           NAMES vigranumpycore.so vigranumpycore.pyd
-          PATHS ${PYTHON_SITE_PACKAGES} 
-                ENV PYTHONPATH
-                ${CMAKE_PREFIX_PATH}/python/Lib/site-packages # Windows, buildem system
+          PATHS ${Python_SITELIB}
           PATH_SUFFIXES vigra)
 
 
-# handle the QUIETLY and REQUIRED arguments and set VIGRA_FOUND to TRUE if 
+# handle the QUIETLY and REQUIRED arguments and set VIGRA_FOUND to TRUE if
 # all listed variables are TRUE
 INCLUDE(FindPackageHandleStandardArgs)
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(VIGRA DEFAULT_MSG VIGRA_IMPEX_LIBRARY VIGRA_INCLUDE_DIR VIGRA_NUMPY_CORE_LIBRARY)
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(
+    VIGRA DEFAULT_MSG
+    VIGRA_IMPEX_LIBRARY
+    VIGRA_INCLUDE_DIR
+    VIGRA_NUMPY_CORE_LIBRARY
+    Python_NumPy_FOUND
+)
 
 MARK_AS_ADVANCED( VIGRA_INCLUDE_DIR VIGRA_IMPEX_LIBRARY VIGRA_IMPEX_LIBRARY_DIR VIGRA_NUMPY_CORE_LIBRARY)
-- 
2.43.0

