From 5f0b1131e2697a1cb2986c74b6f0128a662abfbb Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Wed, 19 Mar 2025 09:47:23 -0400
Subject: [PATCH 6/6] Explicitely link to python libraries with pybind

---
 src/python/lib/CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/python/lib/CMakeLists.txt b/src/python/lib/CMakeLists.txt
index 7b6b47fb..1ae01e18 100644
--- a/src/python/lib/CMakeLists.txt
+++ b/src/python/lib/CMakeLists.txt
@@ -68,9 +68,12 @@ macro(addPythonModule)
     # link additional libraries
     target_link_libraries(_${MODULE_NAME}
         PUBLIC
-        # ${PYTHON_LIBRARIES}
         ${ADD_PY_MOD_LIBRRARIES}
     )
+    target_link_libraries(_${MODULE_NAME}
+        PRIVATE
+        ${Python_LIBRARIES}
+    )
 
     # create module dir
     file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/python/${ADD_PY_MOD_NESTED_NAME})
-- 
2.43.0

