diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1e6985d4..091b66c4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,11 +51,7 @@ find_package(Threads REQUIRED)
 
 
 if(MSVC)
-    # Disable autolinking on MSVC.
-    add_definitions(-DBOOST_ALL_NO_LIB)
     add_definitions(-DNOMINMAX)
-    # Make sure we don't use debug libraries.
-    set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreadedDLL")
 endif()
 
 # from externals..
diff --git a/src/python/CMakeLists.txt b/src/python/CMakeLists.txt
index b4096768..414ce630 100644
--- a/src/python/CMakeLists.txt
+++ b/src/python/CMakeLists.txt
@@ -7,13 +7,6 @@ include_directories(${xtensor-python_INCLUDE_DIRS})
 find_package(pybind11 CONFIG REQUIRED)
 include_directories(${pybind11_INCLUDE_DIRS})
 
-if(MSVC)
-    set(PYBIND11_CPP_STANDARD /std:c++17)
-else()
-    set(PYBIND11_CPP_STANDARD -std=c++17)
-endif()
-message(STATUS "Using c++ 17 for pybind")
-
 add_subdirectory(test)
 add_subdirectory(lib)
 

