diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1e6985d4..091b66c4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,11 +51,7 @@ find_package(Threads REQUIRED)
 
 
 if(MSVC)
-    # Disable autolinking on MSVC.
-    add_definitions(-DBOOST_ALL_NO_LIB)
     add_definitions(-DNOMINMAX)
-    # Make sure we don't use debug libraries.
-    set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreadedDLL")
 endif()
 
 # from externals..
diff --git a/src/python/CMakeLists.txt b/src/python/CMakeLists.txt
index b4096768..414ce630 100644
--- a/src/python/CMakeLists.txt
+++ b/src/python/CMakeLists.txt
@@ -7,13 +7,6 @@ include_directories(${xtensor-python_INCLUDE_DIRS})
 find_package(pybind11 CONFIG REQUIRED)
 include_directories(${pybind11_INCLUDE_DIRS})
 
-if(MSVC)
-    set(PYBIND11_CPP_STANDARD /std:c++17)
-else()
-    set(PYBIND11_CPP_STANDARD -std=c++17)
-endif()
-message(STATUS "Using c++ 17 for pybind")
-
 add_subdirectory(test)
 add_subdirectory(lib)
 
diff --git a/src/python/lib/CMakeLists.txt b/src/python/lib/CMakeLists.txt
index 7b6b47fb..9bf49bca 100644
--- a/src/python/lib/CMakeLists.txt
+++ b/src/python/lib/CMakeLists.txt
@@ -60,15 +60,16 @@ macro(addPythonModule)
     #endif()
     set_target_properties(_${MODULE_NAME}  PROPERTIES PREFIX "")
 
-    #message(STATUS ********Module: ${MODULE_NAME})
-    #message(STATUS ********Libs: ${ADD_PY_MOD_LIBRRARIES})
+    message(STATUS ********Module: ${MODULE_NAME})
+    message(STATUS ********Libs: ${ADD_PY_MOD_LIBRRARIES})
+    message(STATUS ********Sources: ${ADD_PY_MOD_SOURCES})
 
     # NOTE it's not necessary to link against the python library here
     # and if we do, this causes seg-faults on mac, so don't do it!
     # link additional libraries
     target_link_libraries(_${MODULE_NAME}
-        PUBLIC
-        # ${PYTHON_LIBRARIES}
+        PRIVATE
+        ${Python_LIBRARIES}
         ${ADD_PY_MOD_LIBRRARIES}
     )
 

